src/main/scala/is/hail/keytable/KeyTable.scala

```diff
@@ -407,15 +407,17 @@ case class KeyTable(hc: HailContext, rdd: RDD[Row],
            |  Left signature: ${ TStruct(keyFields).toPrettyString(compact = true) }
            |  Right signature: ${ TStruct(other.keyFields).toPrettyString(compact = true) }""".stripMargin)
 
-    val overlappingFields = fieldNames.toSet.intersect(other.fieldNames.toSet) -- key -- other.key
-    if (overlappingFields.nonEmpty)
-      fatal(
-        s"""Columns that are not keys cannot be present in both key tables.
-           |  Overlapping fields: ${ overlappingFields.mkString(", ") }""".stripMargin)
+    val preNames = fieldNames ++ other.valueSignature.fields.map(_.name)
+    val (joinNames, remapped) = mangle(preNames)
```

Comment:
I found the name `joinNames` a bit confusing, these are the de-duplicated names, right?

Reply:
yeah, I'll rename to `finalColumnNames` or something.